src/sentry/api/endpoints/project_events.py

```diff
@@ -39,27 +40,23 @@ def get(self, request, project):
                                      belong to.
         """
 
-        events = Event.objects.filter(
-            project_id=project.id,
-        )
-
         query = request.GET.get('query')
+        conditions = []
         if query:
-            events = events.filter(
-                message__icontains=query,
-            )
-
-        # filter out events which are beyond the retention period
-        retention = quotas.get_event_retention(organization=project.organization)
-        if retention:
-            events = events.filter(
-                datetime__gte=timezone.now() - timedelta(days=retention)
-            )
+            conditions.append(['message', 'LIKE', '%{}%'.format(query)])
+
+        data_fn = partial(
+            # extract 'data' from raw_query result
+            lambda *args, **kwargs: raw_query(*args, **kwargs)['data'],
+            start=timezone.now() - timedelta(days=90),
+            end=timezone.now(),
```

Comment:
Are start and end required (will they not just default to the retention period and the current time?)

Reply:
In the general case yeah it would be much nicer to default them in the snuba util instead of having to do this date stuff everywhere.